src/contrib/persistence/Akka.Persistence.Sql.Common/Journal/QueryExecutor.cs

```diff
@@ -778,27 +778,30 @@ protected DbCommand GetCommand(DbConnection connection, string sql)
         /// <returns>TBD</returns>
         protected virtual void WriteEvent(DbCommand command, IPersistentRepresentation e, IImmutableSet<string> tags)
         {
-            var payloadType = e.Payload.GetType();
-            var serializer = Serialization.FindSerializerForType(payloadType, Configuration.DefaultSerializer);
+            
+            var serializer = Serialization.FindSerializerForType(e.Payload.GetType(), Configuration.DefaultSerializer);
 
             // TODO: hack. Replace when https://github.com/akkadotnet/akka.net/issues/3811
-            string manifest = "";
-            var binary = Akka.Serialization.Serialization.WithTransport(Serialization.System, () =>
+            var (binary,manifest) = Akka.Serialization.Serialization.WithTransport(Serialization.System,(e.Payload,serializer) ,(state) =>
             {
                 
-                if (serializer is SerializerWithStringManifest stringManifest)
+                var (thePayload, theSerializer) = state;
+                string thisManifest = "";
```

Comment:
We're keeping a local here because changing this into a Ternary at end was both:

 - Harder to read
 - Slower because the complex ternary doesn't optimize well =/